tsdb/engine/tsm1/file_store.go

```diff
@@ -258,6 +261,7 @@ func (f *FileStore) Add(files ...TSMFile) {
 	for _, file := range files {
 		atomic.AddInt64(&f.stats.DiskBytes, int64(file.Size()))
 	}
+	f.lastFileStats = nil // Will need to be recalculated on next call to Stats.
```

Comment:
Lower down in the readers, I just resliced this to `f.lastFileStats = f.lastFileStats[:0]` to avoid the extra allocations.  Maybe do that here and check for `len() == 0`?